### PR TITLE
[core] Add Tracer.shutdown helper method

### PR DESCRIPTION
**Pull request by brettlangdon**
Adding a new `Tracer.shutdown` helper to make stopping the background worker easier.

This can also be a place where we can consolidate any other shutdown specific behavior in the future.

### CODE REVIEW

File: ddtrace/tracer.py [hunk 1]
```diff
@@ -604,3 +604,15 @@ def set_tags(self, tags):
         :param dict tags: dict of tags to set at tracer level
         """
         self.tags.update(tags)
+
+    def shutdown(self, timeout=None):
+        """ Shutdown the tracer
```
Comment (jd) [inline ddtrace/tracer.py]: There should not be any leading whitespace, and it should have a final . :)

File: ddtrace/tracer.py [hunk 2]
```diff
@@ -604,3 +604,15 @@ def set_tags(self, tags):
...
+
+        This will stop the background writer/worker and flush any finished traces in the buffer
```
Comment (jd) [inline ddtrace/tracer.py]: final dot

File: ddtrace/tracer.py [hunk 3]
```diff
@@ -604,3 +604,15 @@ def set_tags(self, tags):
...
+
+        :param timeout: How long in seconds to wait for the background worker to flush traces
+            before exiting (default: :obj:`None`)
```
Comment (jd) [inline ddtrace/tracer.py]: maybe explain what `None` means in this context.

File: tests/test_tracer.py [hunk 4]
```diff
@@ -530,3 +530,18 @@ def test_tracer_url():
     with pytest.raises(ValueError) as e:
         t = ddtrace.Tracer(url='foo://foobar:12')
         assert str(e) == 'Unknown scheme `https` for agent URL'
+
+
+def test_tracer_shutdown():
```
Comment (jd) [inline tests/test_tracer.py]: is it really worth it?
It's not really testing anything 🤷‍♂

Comment (brettlangdon) [reply tests/test_tracer.py]: 🤷‍♂ keeps test coverage up 😆

Comment (jd) [reply tests/test_tracer.py]: A good summary why coverage isn't a relevant metric :)

**Comment by Kyle-Verhoog** [comment]
@brettlangdon @jd let me know what you think of the slight change I made to only shutdown the writer thread if it's running 🙂

**Pull request merged**